src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/AbstractCosmosQuery.java

```diff
@@ -27,20 +28,22 @@ public Object execute(Object[] parameters) {
         final ResultProcessor processor = method.getResultProcessor().withDynamicProjection(accessor);
         final String collection = ((CosmosEntityMetadata) method.getEntityInformation()).getCollectionName();
 
-        final CosmosQueryExecution execution = getExecution(accessor);
+        final CosmosQueryExecution execution = getExecution(accessor, processor.getReturnedType());
```

Comment:
When I mentioned for ReactiveCosmosRepository, I actually meant to have this functionality on ReactiveQuery pipeline as well. 
For example, this code needs to be updated in `AbstractReactiveCosmosQuery` as well.